aws_auditing_list.py

```diff
@@ -38,376 +44,678 @@ def default(self, o):
-        count_region = {}
         for region_name in regions:
-            count_region[region_name] = {}
-            count = 0
+            self.ec2_counts[region_name] = {}
             count_list = {}
-
-            conn = self.connect("ec2", region_name=region_name)
+            conn = self.connect_service_region(
+                "ec2", region_name=region_name
+            )
             instance_list = conn.describe_instances()
-            instance_ids = []
-            
-            for r in  instance_list['Reservations']:
+
+            for r in instance_list["Reservations"]:
                 for i in r["Instances"]:
-                    id = i['InstanceId']
-                    instance_ids.append(i['InstanceId'])
+
                     if "KeyName" in i:
                         key_name = i["KeyName"]
                     else:
                         key_name = ""
-                    self.dictionary[region_name]["EC2"][id] = {
-                        "type": i['InstanceType'],
+
+                    self.dictionary[region_name]["EC2"][
+                        id
+                    ] = {
+                        "type": i["InstanceType"],
                         "key_name": key_name,
-                        "launch_time": i['LaunchTime'],
-                        "state": i['State']['Name'],
-                        "volumes": []
+                        "launch_time": i["LaunchTime"],
+                        "state": i["State"]["Name"],
                     }
-
-                    if len(i['BlockDeviceMappings']) > 0:
-                        for vol in i['BlockDeviceMappings']:
-                            self.dictionary[region_name]["EC2"][id]["volumes"].append(vol['Ebs']['VolumeId'])
-                    
-                    instance_type = i['InstanceType']
-                    if i['State']['Name'] == 'running':
-                        if not instance_type in count_list:
-                            count_list[instance_type] = {"count" : 0}
-
+                    instance_type = i["InstanceType"]
+                    if i["State"]["Name"] == "running":
                         if instance_type in count_list:
-                            count_list[instance_type]["count"] += 1
-            count_region[region_name] = count_list
-        self.ec2_counts = count_region        
-
-    def get_classicELB_resources(self, regions):
-        elb_cost_region = {}
-        with open("elb_pricing.json","r") as elb:
+                            count_list[instance_type][
+                                "count"
+                            ] += 1
+
+                        else:
+                            count_list[instance_type] = {
+                                "count": 0
+                            }
+            self.ec2_counts[region_name] = count_list
+
+    def get_classic_elb_resources(self, regions):
+        with open("elb_pricing.json", "r") as elb:
             elb_pricing = json.load(elb)
```

Comment:
I would like this script to take an argument that generates these files if required or errors if they don't exist and need to be generated.